src/lib/routing-tables.js

```diff
@@ -157,53 +152,59 @@ class RoutingTables {
    * the way to `ledgerC`.
    * This connector must have `[ledgerA, ledgerB]` as a pair.
    *
-   * @param {URI} ledgerA
-   * @param {URI} ledgerC
+   * @param {IlpAddress} ledgerA
+   * @param {IlpAddress} ledgerC
    * @param {String} finalAmount
    * @returns {Object}
    */
   findBestHopForDestinationAmount (ledgerA, ledgerC, finalAmount) {
     const nextHop = this._findBestHopForDestinationAmount(ledgerA, ledgerC, +finalAmount)
     if (!nextHop) return
+    // ledgerA is not necessarily the sourceLedger because the latter is the
+    // longest known prefix of the former (likewise for ledgerC/finalLedger).
```

Comment:
It means I should probably rename `ledgerA` to `sourceAddress` and `ledgerC` to `finalAddress` so that the variable names make sense again.

In other words, if you call `findBestHopForDestinationAmount("ilpdemo.red.alice", "ilpdemo.blue.bob", "100.00")`, the comment is saying that `"ilpdemo.red."` is the source ledger, not `"ilpdemo.red.alice"` (and likewise for the final ledger).